### ISSUE TITLE
Don't write bundles to node_modules and make this behaviour the default

### ISSUE THREAD

**Issue by eqyiel** [keystonejs/keystone#4436]
I realise that there's `KEYSTONE_PREBUILD_ADMIN=true` but it's weird and breaks determinism.  It also makes it difficult to package `keystone` in [Nix](https://nixos.org/nix/), where immutability is enforced.

Without a bunch of hacks to pre-build keystone, the status of using this with Nix right now is that the admin bundles are re-generated _every time_ an admin page is requested because `/nix/store` is read only.

It would be great to be able to choose the location that these files are written to.

Edit: accidental keyboard mashing

**Comment by eqyiel** [comment]
Alternatively, the built bundles could be published on npm along with keystone.

I've put up an example of the workaround I'm using here, which does pretty much that: https://gist.github.com/eqyiel/e1d2cb3cee89a581dcd06fdab40e2ecc

**Comment by eqyiel** [comment]
This may be a misunderstanding.  It looks like the bundles _are_ published on npm, but aren't in this repo (which makes sense).

What I'm emphatically trying to avoid is having any transpiling done on the server when an admin route is requested, which is the behaviour I see when running without `KEYSTONE_PREBUILD_ADMIN=true`.

Is that the purpose of this environment variable?

**Comment by Noviny** [comment]
@eqyiel That's the purpose of that environment variable. The admin UI's build is in part based on the models specific to the project, so keystone at very least needs to rebuild/verify them on startup. As much as possible, we are frontloading this creation to the keystone startup process, rather than runtime.

Is the problem that this process isn't deterministic, or that changing models leads to build changes?

**Comment by eqyiel** [comment]
@Noviny the auto generated admin UI is one of the things that drew my team to Keystone in the first place, it's just that the bundles get written to `node_modules`, which is unexpected.

And one some systems like NixOS, it's read-only.

What I'm suggesting is that those bundles get created elsewhere, or at least that the path to where the generated admin bundles are saved is configurable.

We're still taking baby steps with Keystone, but if we get to the point of using this in production one of us will probably step up with a PR. 😄

**Comment by Noviny** [comment]
@eqyiel Yeah, can see where it would be useful to be able to specify that. It's not a use-case we've had to account for, but I'm sure if you had a PR for it, we'd be happy to help you get it in. 😃 

We're looking to move feature requests out of issues for neatness, and setting up a github project for it. I've added this request to that and am closing this. Note that the names on the project are very in progress.
